internal/provider/kubernetes/status_updater.go

```diff
@@ -154,13 +160,18 @@ type Updater interface {
 
 // UpdateWriter takes status updates and sends these to the UpdateHandler via a channel.
 type UpdateWriter struct {
+	enabled       <-chan struct{}
 	updateChannel chan<- Update
 }
 
 // Send sends the given Update off to the update channel for writing by the UpdateHandler.
 func (u *UpdateWriter) Send(update Update) {
 	// Non-blocking receive to see if we should pass along update.
-	u.updateChannel <- update
+	select {
```

Comment:
This line can block wtaching

```
u.updateChannel <- update
```

Reply:
It seems that we got regression for https://github.com/envoyproxy/gateway/issues/4336

https://github.com/envoyproxy/gateway/actions/runs/11965697450/job/33361599433?pr=4754
Messages:     error waiting for Gateway status to have listeners matching expectations

An alternative approach is decouping predicate and gateway status update: https://github.com/envoyproxy/gateway/pull/4767